packages/nars/src/Static.tsx

```diff
@@ -1,98 +1,112 @@
 import * as React from "react";
 import {
+  Prop,
   ComponentConfig,
-  ExtractInputPropType,
-  LocalPropKey,
+  Props,
   PropType,
+  isSuccess,
+  RPCInterface,
+  LocalPropType,
+  sanitizeAndEncode,
+  Arguments,
 } from "nars-common";
 import { startListening, server as Server } from "./NarsServer.gen";
-import { t as LocalProp } from "./LocalProp.gen";
 import { Dict_t } from "./shims/Js.shim";
 import { t as JsValue_t } from "./JsValue.gen";
+import {
+  ExtractPropType,
+  ExtractPropTypes,
+  ComponentDefinitions,
+  Decoder,
+} from "./StaticPropTypes";
+import { toJs } from "./RpcInterface.gen";
 
-export type ExtractPropTypes<T> = {
-  [K in keyof T]: K extends string
-    ? T[K] extends LocalPropKey<any, any, any>
-      ? LocalProp
-      : ExtractInputPropType<T[K]>
-    : never;
-};
-
-export type ComponentDefinitions<T extends ComponentConfig> = {
-  readonly [P in keyof T]: React.ComponentType<ExtractPropTypes<T[P]>>;
-};
-
-type Decoder<T extends ComponentConfig> = {
-  readonly [P in keyof T]: (
-    props: Dict_t<JsValue_t> | undefined,
-    localPropKeys: string[]
-  ) => ExtractPropTypes<T[P]>;
-};
-
-type Writeable<T> = { -readonly [P in keyof T]: T[P] };
+function map<
```

Comment:
Maybe it'd be possible to make it more well-typed if we passed propsIn to this function, too.